colossalai/checkpoint_io/utils.py

```diff
@@ -139,6 +140,8 @@ def shard_optimizer_checkpoint(state_dict: dict, max_shard_size: int = 1024) ->
         state_size = 0
         isDTensor = False
         for state_tensor in state.values():
+            if state_tensor is None:
```

Comment:
In which case will the `state_tensor` be None?

Reply:
The state 'momentum_buffer' of SGD optimizer can be None. In my case if I delete this two lines, the code will fail the test for ddp_checkpoint & fsdp_checkpoint.